internal/gatewayapi/route.go

```diff
@@ -227,12 +227,20 @@ func (t *Translator) processHTTPRouteRules(httpRoute *HTTPRouteContext, parentRe
 
 func processTimeout(irRoute *ir.HTTPRoute, rule gwapiv1.HTTPRouteRule) {
 	if rule.Timeouts != nil {
+		var rto *ir.Timeout
+
+		if irRoute.Timeout != nil {
```

Comment:
nit: policy is computed/translated post route translation, so we can assume its empty here

Reply:
I wanted to this to be safe in case there's a re-arrangement of the computation order. If you feel that we can make this assumption, i'll update the implementation accordingly.

Reply:
sure, if thats the case, can you also add a comment here, so the next dev can understand why this done this way similar to the comment you have in policy around request timeout